x-pack/plugins/infra/public/containers/logs/log_analysis/api/validate_datasets.ts

```diff
@@ -0,0 +1,36 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import {
+  LOG_ANALYSIS_VALIDATE_DATASETS_PATH,
+  validateLogEntryDatasetsRequestPayloadRT,
+  validateLogEntryDatasetsResponsePayloadRT,
+} from '../../../../../common/http_api';
+import { decodeOrThrow } from '../../../../../common/runtime_types';
+import { npStart } from '../../../../legacy_singletons';
+
+export const callValidateDatasetsAPI = async (
+  indices: string[],
+  timestampField: string,
+  startTime: number,
+  endTime: number
+) => {
+  const response = await npStart.http.fetch(LOG_ANALYSIS_VALIDATE_DATASETS_PATH, {
```

Comment:
Can this be changed so as to not use the `legacy_singletons`, e.g. `callValidateDatasetsAPI` taking `fetch` as an argument?

Reply:
Absolutely - I just kept it in sync with the other API calls so far. Would it be ok to switch all of them over at once in a separate PR (that is not subject to a close deadline)?

Reply:
Yep - of course 👍